rs/xnet/payload_builder/src/lib.rs

```diff
@@ -1602,7 +1569,13 @@ impl XNetClient for XNetClientImpl {
         // TODO(MR-28) Make timeout configurable.
         let result = tokio::time::timeout(Duration::from_secs(5), async {
             let request_start = Instant::now();
-            let result = self.http_client.get(endpoint.url.clone()).await;
+
+            let response = self
+                .http_client
+                .get(endpoint.url.clone())
+                .await
+                .map_err(XNetClientError::RequestFailed)?;
```

Comment:
What about timeouts? We don't seem to produce the `XNetClientError::Timeout` variant anymore.

Reply:
I think it is not needed here because if the get request takes too long the `tokio::time::timeout` wrapper will catch it and report it on line 1601

Reply:
Oh, thanks. I did search for `XNetClientError::Timeout` and somehow didn't find the occurrence on line 1601.

Then, assuming `self.http_client` doesn't produce any timeouts of its own (which seems likely, we don't set an explicit timeout and any default would be significantly higher than 5 seconds), this looks fine.